packages/core/factories/src/useUserAddressFactory.ts

```diff
@@ -0,0 +1,62 @@
+import { UseUserAddress, UserAddressType } from '@vue-storefront/interfaces';
+import { Ref, ref, computed } from '@vue/composition-api';
+
+export type UseUserAddressFactoryParams<ADDRESS, ADDRESS_TYPE = UserAddressType> = {
+  addresses: ADDRESS[];
+  searchAddresses: (params?: {
+    [x: string]: any;
+  }) => Promise<ADDRESS[]>;
+  addAddress: (address: ADDRESS, type: ADDRESS_TYPE) => Promise<void>;
+  deleteAddress: (address: ADDRESS) => Promise<void>;
+  updateAddress: (address: ADDRESS) => Promise<void>;
+  getBillingAddresses: () => ADDRESS[];
+  getShippingAddresses: () => ADDRESS[];
+}
+
+export function useUserAddressFactory<ADDRESS, ADDRESS_TYPE = UserAddressType> (factoryParams: UseUserAddressFactoryParams<ADDRESS, ADDRESS_TYPE>) {
+  const loading: Ref<boolean> = ref(false);
+
+  const searchAddresses = async (params) => {
+    loading.value = true;
+    await factoryParams.searchAddresses(params).then(addresses => {
+      factoryParams.addresses.length = 0;
+      addresses.forEach(address => factoryParams.addresses.push(address));
+    });
+    loading.value = false;
```

Comment:
please change that to try/catch/finally blocks for this :)
and why isn't that just `factoryParams.addresses.value = addresses`?

Reply:
Oh, that's because Ref works really bad with arrays (or just doesn't work at all)